egui/src/menu.rs

```diff
@@ -120,37 +120,44 @@ pub(crate) fn menu_ui<'c, R>(
     ctx: &CtxRef,
     menu_id: impl std::hash::Hash,
     menu_state_arc: &Arc<RwLock<MenuState>>,
-    mut style: Style,
     add_contents: impl FnOnce(&mut Ui) -> R + 'c,
 ) -> InnerResponse<R> {
     let pos = {
         let mut menu_state = menu_state_arc.write();
         menu_state.entry_count = 0;
         menu_state.rect.min
     };
-    // style.visuals.widgets.active.bg_fill = Color32::TRANSPARENT;
-    style.visuals.widgets.active.bg_stroke = Stroke::none();
-    // style.visuals.widgets.hovered.bg_fill = Color32::TRANSPARENT;
-    style.visuals.widgets.hovered.bg_stroke = Stroke::none();
-    style.visuals.widgets.inactive.bg_fill = Color32::TRANSPARENT;
-    style.visuals.widgets.inactive.bg_stroke = Stroke::none();
+
     let area = Area::new(menu_id)
         .order(Order::Foreground)
         .fixed_pos(pos)
         .interactable(false)
         .drag_bounds(Rect::EVERYTHING);
-    let frame = Frame::menu(&style);
     let inner_response = area.show(ctx, |ui| {
-        frame
-            .show(ui, |ui| {
-                const DEFAULT_MENU_WIDTH: f32 = 150.0; // TODO: add to ui.spacing
-                ui.set_max_width(DEFAULT_MENU_WIDTH);
-                ui.set_style(style);
-                ui.set_menu_state(Some(menu_state_arc.clone()));
-                ui.with_layout(Layout::top_down_justified(Align::LEFT), add_contents)
-                    .inner
-            })
-            .inner
+        ui.scope(|ui| {
+            let style = ui.style_mut();
+            style.spacing = Spacing {
+                item_spacing: Vec2::ZERO,
+                button_padding: crate::vec2(2.0, 0.0),
+                ..Default::default()
+            };
```

Comment:
perhaps we should only over-write `item_spacing` and `button_padding` and leave the rest of the fields?

Reply:
I agree. Though they should probably not be hardcoded at all. At least not the `button_padding`. Would adding a`spacing.menu_button_padding` be appropriate?